docs/Deploy HotStack on  PSI.md

```diff
@@ -0,0 +1,226 @@
+	HotStack's functionality is modularized through several key Ansible roles :
+
+	- **dataplane_ssh_keys**: Responsible for generating SSH keys essential for dataplane communication and Nova instance migration processes.
+    
+	- **heat_stack**: This role is fundamental for interacting with the underlying OpenStack cloud. It deploys the necessary infrastructure components (virtual machines, networks, etc.) by orchestrating an OpenStack Heat template.
+    
+	- **ocp_agent_installer**: Manages the installation of the OpenShift Container Platform, specifically using an agent-based installation method which often involves PXE booting.
+    
+	- **controller**: This role handles the post-provisioning setup of the designated controller node. Its tasks include waiting for the node to become available in the Ansible inventory, ensuring SSH reachability, and executing bootstrap configurations.
+    
+	- **hotloop**: Provides a generic looping mechanism to execute sequences of commands, apply Kubernetes manifests (Custom Resources - CRs), and implement wait conditions, facilitating complex automation workflows.
+    
+	- **redfish_virtual_bmc**: Deploys the sushy-emulator, a RedFish Virtual BMC (Baseboard Management Controller) service, typically onto the OpenShift cluster. This is often crucial for managing virtualized "bare metal" nodes as required by RHOSO components like OpenStack Ironic.
+    
+
+* **Key Ansible Playbooks**:
+	The deployment process is orchestrated by a sequence of Ansible playbooks, with bootstrap.yml serving as the main entry point. This master playbook imports other playbooks to execute specific stages of the deployment:
+
+	1. **01-infra.yml**: Provisions the virtual infrastructure on the OpenStack cloud.
+	2. **02**-**bootstrap_controller**.yml: Prepares and bootstraps the controller node.
+	3. **03**-**install_ocp**.yml: Installs the OpenShift Container Platform cluster.
+	4. **04_redfish_virtual_bmc**.yml: Deploys the sushy-emulator (RedFish Virtual BMC).
+	5. **05_deploy_rhoso**.yml: Deploys the Red Hat OpenStack overcloud (RHOSO).
+
+---
+# 2. Preparing Your OpenStack Environment and Client Machine
+
+- Successful HotStack deployment depends on OpenStack cloud environment and the client machine from which deployment will be orchestrated.
+- It is recommended to spin up the small client instance which will be used for deployment.
+- So we have to meticulous steps to meet prerequisites or else it will lead to deployment failure.
+
+## 2.1 Client-Machine
+- It is recommended to create a small instance on OpenStack Cloud(PSI) to run the deployment script.
+- Recommended flavor and OS: 
+	- g.standard.small
+	- CentOS-Stream-9-latest
+## 2.2 Client-Side Tooling (Machine Running Ansible)
+After SSH into Client Machine we need tools to run the OpenStack
+
+* **Ansible:** A HotStack is built upon Ansible, a working Ansible installation is mandatory.
+	* One Way to install ansible is:
+```
+pip3 install ansible
```

Comment:
I prefer using RPMs `sudo dnf install ansible-core` 
May want to add two examples, one with pip and one with dnf package manager?

Also, if using pip, we should use a virtual env so:
1. mkdir -p ~/ansible-venv
2. python4 -m venv ~/ansible-venv
3. source ~/ansible-venv/bin/activate
4. pip install ansible